File: loadbalancer.rb

require 'socket'

host = "192.168.79.130"
port = 1500
loadbalancer = TCPServer.open(host,port)

while true
client = loadbalancer.accept

loadhost=client.peeraddr.last
puts "#{loadhost} connect to loadbalancer"

fromclient=client.gets
puts "Message from client : #{fromclient}"

if fromclient=="friend"
sockettoserver1 = TCPSocket.open("192.168.223.128",3000)
sockettoserver1.write(fromclient)
sockettoserver1.close

else if fromclient=="enemy"
sockettoserver2 = TCPSocket.open("192.168.223.131",4000)
sockettoserver2.write(fromclient)
sockettoserver2.close

else
puts "ERROR"

end
end
end

File: loadbalancer.java
import java.io.*;
import java.net.*;

public class loadbalancer{

	public static void main( String []args) throws IOException{
	ServerSocket s1 = new ServerSocket (1234);
	while (true)
	{
	
	Socket connectionsocket = s1.accept();
	BufferedReader messagefromclient = new BufferedReader( new InputStreamReader (connectionsocket.getInputStream()));
	String cs = messagefromclient.readLine();
	
	if(cs.contains("enemy"))
	{
		
		Socket server = new Socket("192.168.223.128",2345);
		DataOutputStream outtoserver = new DataOutputStream( server.getOutputStream());
		outtoserver.writeBytes(cs + "\n");
		server.close();
	}
		
	else if(cs.contains("friend"))
	{

		Socket server = new Socket("192.168.223.131",2346);
		DataOutputStream outtoserver = new DataOutputStream( server.getOutputStream());
		outtoserver.writeBytes(cs + "\n");
		server.close();
	}

	else
		
		System.out.println("Error");
	}
}
}	
